resolver/src/resolver.rs

```diff
@@ -78,7 +78,16 @@ impl Resolver {
     ///
     /// A new Resolver or an error if there was an error with the configuration.
     pub fn new(config: ResolverConfig, options: ResolverOpts) -> io::Result<Self> {
-        let lru = Arc::new(Mutex::new(DnsLru::new(options.cache_size)));
+        let mut lru = DnsLru::new(options.cache_size);
+        if let Some(min) = options.min_positive_ttl {
+            lru = lru.with_min_positive_ttl(min);
+        }
+        if let Some(min) = options.min_negative_ttl {
+            lru = lru.with_min_negative_ttl(min);
+        }
```

Comment:
NIT: We just put this duplicated logic into `DnsLru::new()` (it's duplicated in ResolverFuture).

Reply:
fef5f93c11a6e46e2e4e5aae5245ea00e0e3de2b also makes this bit a little less ugly :)